File: src/server/coordinates.py
class Coordinate:
    def __init__(self, lat: float, lng: float):
        self.lat = lat
        self.lng = lng

    def to_tuple(self):
        return (self.lat, self.lng)

    def to_lnglat(self):
        return (self.lng, self.lat)  # ORS format

File: src/client/modules/coordinates.ts
// src/utils/coordinates.ts

/** Represents a single coordinate (latitude, longitude) */
export class Coordinate {
  lat: number;
  lng: number;

  constructor(lat: number, lng: number) {
    this.lat = lat;
    this.lng = lng;
  }

  /** Return as a tuple [lat, lng] */
  toTuple(): [number, number] {
    return [this.lat, this.lng];
  }

  /** Optional: string representation */
  toString(): string {
    return `${this.lat}, ${this.lng}`;
  }
}

/** Type alias for an array of coordinates representing a polygon or path */
export type CoordinateArray = Coordinate[];
